insights/parsers/systemd/unitfiles.py

```diff
@@ -81,3 +93,76 @@ def is_on(self, service_name):
             bool: True if service is enabled, False otherwise
         """
         return self.services.get(service_name, False)
+
+
+@parser('systemctl_list-units')
+class ListUnits(Parser):
+
+    """
+    The ListUnits class parses the output of ``/bin/systemctl list-units`` and provides
+    information about all the services listed under it.
+
+    Output of Command::
+
+        sockets.target                      loaded active active    Sockets
+        swap.target                         loaded active active    Swap
+        systemd-shutdownd.socket            loaded active listening Delayed Shutdown Socket
+        neutron-dhcp-agent.service          loaded active running   OpenStack Neutron DHCP Agent
+        neutron-openvswitch-agent.service   loaded active running   OpenStack Neutron Open vSwitch Agent
+
+    Example:
+
+        >>> conf = shared[ListUnits]
+        >>> conf.get_service_details('swap.target')
+        {'LOAD': 'loaded', 'ACTIVE': 'active', 'SUB': 'active', 'UNIT': 'swap.target'}
+    """
+    def __init__(self, *args, **kwargs):
+        self.unit_list = {}
```

Comment:
@vishwanathjadhav it would be helpful to show usage of the `unit_list` attribute in the examples assuming you intend for it to be public.